core/src/test/java/org/apache/calcite/sql/type/SqlTypeUtilTest.java

```diff
@@ -197,4 +198,35 @@ private RelDataType struct(RelDataType...relDataTypes) {
     }
     return builder.build();
   }
+
+  private void compareTypesIgnoringNullability(
+      RelDataType type1, RelDataType type2, boolean expectedResult) {
+    String msg = String.format(Locale.ROOT,
+        "%s and %s are expected to be %s ignoring nullability.",
+        type1.getFullTypeString(),
+        type2.getFullTypeString(),
+        expectedResult ? "equal" : "not equal");
+
+    assertThat(msg,
+        SqlTypeUtil.equalSansNullability(f.typeFactory, type1, type2), is(expectedResult));
+    assertThat(msg,
+        SqlTypeUtil.equalSansNullability(type1, type2), is(expectedResult));
+  }
+
+  @Test public void testEqualSansNullability() {
+    RelDataType bigIntType = f.sqlBigInt;
+    RelDataType nullableBigIntType = f.sqlBigIntNullable;
+    RelDataType varCharType = f.sqlVarchar;
+    RelDataType bigIntType1 =
+        f.typeFactory.createTypeWithNullability(nullableBigIntType, false);
+
+    // different types
```

Comment:
This should a parameter to `compareTypesIgnoringNullability` method.

Reply:
I guess you mean the type factory should be a parameter? Revised accordingly. Thanks.

Reply:
I mean `different types` should be a comment parameter rather than a comment

Reply:
Revised. Thanks for your clarification.